streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SettingServiceImpl.java

```diff
@@ -148,8 +153,24 @@ public ResponseResult checkDocker(DockerConfig dockerConfig) {
   }
 
   @Override
-  public ResponseResult checkEmail(SenderEmail senderEmail) {
-    // TODO check
+  public ResponseResult checkEmail(SenderEmail senderEmail) throws MessagingException {
```

Comment:
Do not throw exceptions to the Controller layer, return the validation result instead.

```
 @Override
    public ResponseResult checkEmail(SenderEmail senderEmail) {
        ResponseResult result = new ResponseResult();
        Properties props = new Properties();
        props.put("mail.smtp.auth", "true");
        props.put("mail.smtp.starttls.enable", "true");
        props.put("mail.smtp.host", senderEmail.getHost());
        props.put("mail.smtp.port", senderEmail.getPort());

        Session session = Session.getInstance(props);
        try {
            Transport transport = session.getTransport("smtp");
            transport.connect(
                senderEmail.getHost(), senderEmail.getUserName(), senderEmail.getPassword());
            transport.close();
            result.setStatus(200);
        } catch (MessagingException e) {
            result.setStatus(500);
            result.setMsg("connect to target mail server failed: " + e.getMessage());
        }
        return result;
    }
```